Security Consideration Extension (#1455)
moq-wg/moq-transport: draft-ietf-moq-transport.md

{{model-object}}).

This extension can be added by the Original Publisher, but MUST NOT be added by
relays. This extension MUST NOT be modified or removed.

# Security Considerations {#security}

MOQT is a protocol used hop-by-hop in a chain between original
publishers to relay, relay to relay, and relay to end
subscribers. Thus, the security consideration need to consider first
what happens between two nodes, but also consider the impacts end to
end over several hops of MOQT.

This results in a trust model where on each hop the nodes needs to be
securely identified, authorized to use resources of the peer, provide
confidentiality and integrity to prevent third party attacks and limit
monitoring and leakage of privacy sensitive information. The relays

> within the chain from origianal publisher to end subscribers will have

Reviewer: ```suggestion
within the chain from original publisher to end subscribers will have
```

an access to media track identifiers as well as the media content
unless it is end-to-end encrypted {{sec-media}}.

TODO: Expand this section, including subscriptions.

TODO: Describe Cache Poisoning attacks

## Communication Security

MOQT depends on a secure transport to provide confidentiality,
integrity and endpoint authentication between subscriber and
publisher. Implementations use QUIC or WebTransport that both fulfill
the basic communication security requirements and these
implementations SHOULD follow best practices for TLS 1.3 and QUIC
(certificate validation, supported cipher suites, OCSP/CRL checks or
short-lived certificates, etc.). Relay authentication is required to
prevent impersonation of relays.

Authentication of original publishers and end subscribers will not
depend on TLS based mechanisms due to challenges with identifiers and
certificate distribution, although mutual TLS MAY be used when
appropriate in the protocol deployment. See {{sec-authorization}} for
discussion of how authorization of original publishers and end
subscribers is achived.

It must be noted that the basic security protection offered by QUIC or

> TCP/TLS does not prevent traffic pattern analysis as media object

Reviewer 2: ```suggestion
TCP/TLS does not prevent traffic pattern analysis as object
```

{{model-object}}).

This extension can be added by the Original Publisher, but MUST NOT be added by
relays. This extension MUST NOT be modified or removed.

# Security Considerations {#security}

TODO: Expand this section, including subscriptions.

TODO: Describe Cache Poisoning attacks

## Communication Security

MOQT depends on a secure transport with confidentiality, integrity and
endpoint authentication between subscriber and
publisher. Implementations use QUIC or WebTransport that both fulfill
the basic communication security requirements and these
implementations SHOULD follow best practices for TLS 1.3 and QUIC
(certificate validation, supported cipher suites, OCSP/CRL checks or
short-lived certificates, etc.). Server authentication is required to
prevent impersonation of publishers or relays.

It must be noted that the basic security protection offered by QUIC or
TCP/TLS does not prevent traffic pattern analysis as media object
sizes, sizes of request messages etc can make it possible for an
observer of the traffic between subscriber and publisher to identify

> media content, user patterns and analyse media stream origin.

Author: I assume you mean encrypted, just source authentication does nothing for middle node. I guess this text may need a high level security model description with the partial trust in relays.

## Authorization

Subscriber authorization to use relays and access published media
tracks are primarily fullfilled using security token based schemes,
although mutual TLS would be possible to use in some deployments where
one have the certificate deployment in place. Mutual TLS only
allows the publisher to identify the directly connected subscriber,
thus make it mostly useful for relay to verify their peer relays.

MOQT has functionality to carry Authorization tokens as message
parameters when requests are sent. These tokens can be of type the
service requires to meet its security goals which can be
varied. Therefore, already two variants of authorization tokens have
been defined for MOQT and more may be defined in the future. The
current tokens are Privacy Pass Authentication for Media over QUIC
{{PPA}} and Authentication scheme for MOQT using Common Access Tokens
{{CAT}}.

## Media Security

The media content transported by MOQT over various tracks from various

> original publishers are subject to several consideration. The first is

Reviewer 3: ```suggestion
original publishers are subject to several considerations. The first is
```

end over several hops of MOQT.

This results in a trust model where on each hop the nodes need to be
securely identified, authorized to use resources of the peer, provide
confidentiality and integrity to prevent third party attacks and limit
monitoring and leakage of privacy sensitive information. The relays
within the chain from original publisher to end subscribers will have
access to track identifiers as well as the object's content
unless it is end-to-end encrypted {{sec-media}}.

TODO: Expand this section, including subscriptions.

TODO: Describe Cache Poisoning attacks

## Communication Security

MOQT depends on a secure transport to provide confidentiality,
integrity and endpoint authentication between subscriber and
publisher. Implementations use QUIC or WebTransport to fulfill
the basic communication security requirements and these
implementations SHOULD follow best practices for TLS 1.3 and QUIC.
Relay authentication is required to
prevent impersonation of relays.

Authentication of original publishers and end subscribers does not
depend on TLS based mechanisms due to challenges with identifiers and
certificate distribution, although mutual TLS MAY be used when

> appropriate  {{sec-authorization}}.

Reviewer 2: ```suggestion
```

It must be noted that the basic security protection offered by QUIC or
TCP/TLS does not prevent traffic pattern analysis as object
sizes, sizes of request messages etc can make it possible for an third
party observer of the traffic between subscriber and publisher to
identify media content, user patterns and media stream origin.

## Authorization {#sec-authorization}

Authentication of original publishers and end subscribers does not
depend on TLS based mechanisms due to challenges with identifiers and
certificate distribution. Instead they rely primarily on

> token based schemes.

Reviewer 2: ```suggestion
Authentication of original publishers and end subscribers does not
depend on TLS based mechanisms due to challenges with identifiers and
certificate distribution, although mutual TLS MAY be used when
appropriate {{sec-authorization}}. Instead authentication relies
primarily on token based schemes.
```

Relays will be identified using TLS based authentication to prevent
impersanation attacks. Mutual TLS is expected to be widely used for
node level identification between relays, especially within one
organization. However, in some deployments mutual TLS may be possible
to use also for end subscribers or original publishers. However, as
only node level authentication is provided, what a particular
identified node is allowed to do is not provided at TLS level.

MOQT has functionality to carry Authorization tokens as message
parameters when requests are sent. These tokens can be of type the
service requires to meet its security goals which can be
varied. Therefore, already two variants of authorization tokens have
been defined for MOQT and more may be defined in the future. The
current tokens are Privacy Pass Authentication for Media over QUIC
{{PPA}} and Authentication scheme for MOQT using Common Access Tokens

> {{CAT}}.

Reviewer 2: ```suggestion
MOQT can carry Authorization tokens as message parameters.
The details of these tokens are chosen by a service to meet its security
requirements. Two variants of authorization tokens have been defined for MOQT:
Privacy Pass Authentication for Media over QUIC {{PPA}} and
Authentication scheme for MOQT using Common Access Tokens {{CAT}}.
```

The tokens is expected to contain information about which actions and
which resources the end point providing the token is authorized to
perform and access. Relays when receiving requests will verify the

> token to determine that the request made is authorized.

Reviewer 2: ```suggestion
Tokens are expected to contain information about which actions and
resources the endpoint providing the token is authorized to perform and
access. When receiving requests, relays verify the token to determine
if the request is authorized.
```

is source authenticity, i.e. to know that the received media objects
are what the original publisher actually published. The information
needing to be source authenticated is not only the encoded media
itself, but also some meta data. For example timestamps are cruical to
understand where on the timeline this media fragment belongs.

The second aspect is content confidentiality, i.e. how to prevent the
relays from having access to the media content itself and minimize what
the relay can deduce about the content. Media object
sizes and traffic patterns do enable some analysis of the
content. Also the track namespace and track name is possible to
analyze and also correlate between end subscribers by relays.

The end-to-end media security are handled by mechanisms external to this
specification. They need to provide source authenticity and
confidentiality. MOQT's media object model do enable both the media
data itself as well as media object headers to be confidentiality and
integrity protected, as well a supporting a class of visible to relay
headers that are integrity protected but not encrypted.

Current proposals for media security include:

>  - An E2EE scheme based on SFRAME: {{I-D.jennings-moq-secure-objects}}.

Reviewer 2: ```suggestion
MOQT's object model enables both the Object data itself as well
as Object Properties to be confidentiality and integrity protected.
MoQT Object Properties can also be integrity protected without
encryption.

Current proposals for track security include:
 - An E2EE scheme based on SFRAME: {{I-D.jennings-moq-secure-objects}}.
```
